script.js

```diff
@@ -1,4 +1,317 @@
+  var output = "try again";
+  var totalResults = 3;
+  var randomResult = rollNumber(totalResults);
+  if (randomResult == 1) {
+    output = "paper";
+  }
+  if (randomResult == 2) {
+    output = "scissors";
+  }
+  if (randomResult == 3) {
+    output = "rock";
+  }
+  return output;
+};
+
+// Choice of Game Mode
+var gameMode = function (input) {
+  if (input == "scissors, paper, stone") {
+    currentGameMode = "scissors, paper, stone"; // normal SPS
+  }
+  if (input == "reverse SPS") {
+    currentGameMode = "reverse SPS"; // SPS where the rules are reversed
+  }
+  if (input == "reverse reverse SPS") {
+    currentGameMode = "scissors, paper, stone"; // reverse reverse SPS so it's normal SPS
+  }
+  if (input == "Korean SPS") {
+    currentGameMode = "Korean SPS"; // korean SPS
+  }
+  if (input == "Computer") {
+    currentGameMode = "Computer";
+  }
+  return "you have chosen " + currentGameMode;
+};
+
+// Reversed Game Mode
+var gameResultRSPS = function (input, enemyOutput) {
+  if ((input == scissors || input == rScissors) && enemyOutput == rock) {
+    myOutputValue = "you won " + message(username, winCount, loseCount);
+    winCount = winCount + 1;
+  }
+  if ((input == scissors || input == rScissors) && enemyOutput == scissors) {
+    myOutputValue = "draw! " + message(username, winCount, loseCount);
+    drawCount = drawCount + 1;
+  }
+  if ((input == scissors || input == rScissors) && enemyOutput == paper) {
+    myOutputValue = "you lose " + message(username, winCount, loseCount);
+    loseCount = loseCount + 1;
+  }
+  if ((input == paper || input == rPaper) && enemyOutput == rock) {
+    myOutputValue = "you lose " + message(username, winCount, loseCount);
+    loseCount = loseCount + 1;
+  }
+  if ((input == paper || input == rPaper) && enemyOutput == scissors) {
+    myOutputValue = "you won " + message(username, winCount, loseCount);
+    winCount = winCount + 1;
+  }
+  if ((input == paper || input == rPaper) && enemyOutput == paper) {
+    myOutputValue = "draw! " + message(username, winCount, loseCount);
+    drawCount = drawCount + 1;
+  }
+  if ((input == rock || input == rRock) && enemyOutput == rock) {
+    myOutputValue = "draw! " + message(username, winCount, loseCount);
+    drawCount = drawCount + 1;
+  }
+  if ((input == rock || input == rRock) && enemyOutput == scissors) {
+    myOutputValue = "you lose " + message(username, winCount, loseCount);
+    loseCount = loseCount + 1;
+  }
+  if ((input == rock || input == rRock) && enemyOutput == paper) {
+    myOutputValue = "you won " + message(username, winCount, loseCount);
+    winCount = winCount + 1;
+  }
+  if (
+    input !== rock &&
+    input !== scissors &&
+    input !== paper &&
+    input !== rRock &&
+    input !== rScissors &&
+    input !== rPaper
+  ) {
+    myOutputValue = "please input rock, paper or scissors"; // input validation.
+  }
+  totalGames = totalGames + 1;
+  console.log(myOutputValue);
+  return myOutputValue;
+};
```

Comment:
well done setting the game logic here, you might want to try using 
if(...){
}else if(...){
}else

for future long chains of if statements